tests/test_downloads_bucket.py

```diff
@@ -0,0 +1,82 @@
+import requests
+
+
+@pytest.fixture(scope="function", name="downloads_bucket_name")
+def fixture_downloads_bucket_name(deployment_tier):
+    suffix = ""
+    if deployment_tier in ["test", "modl"]:
+        suffix = f"-{deployment_tier}"
+    return f"downloads.curibio{suffix}.com"
+
+
+def _create_generic_object(
+    sts_client: STSClient, deployment_account_id: str, downloads_bucket_name: str
+) -> Dict[str, Any]:
+    assumed_role_object = sts_client.assume_role(
+        RoleArn=f"arn:aws:iam::{deployment_account_id}:role/s3_downloads_role",
+        RoleSessionName="a-role-session-name",
+    )
+    credentials = assumed_role_object["Credentials"]
+    s3_resource = boto3.resource(
+        "s3",
+        aws_access_key_id=credentials["AccessKeyId"],
+        aws_secret_access_key=credentials["SecretAccessKey"],
+        aws_session_token=credentials["SessionToken"],
+    )
+    bucket = s3_resource.Bucket(downloads_bucket_name)
+    obj_key = f"tests/{uuid.uuid4()}"
+    obj_contents = str(uuid.uuid4())
+
+    created_object = bucket.put_object(Body=obj_contents, Key=obj_key)
+    return {"object": created_object, "contents": obj_contents, "key": obj_key}
+
+
+@pytest.fixture(scope="function", name="Given_an_object_is_in_the_downloads_bucket")
+def fixture_Given_an_object_is_in_the_downloads_bucket(
+    deployment_aws_account_id, downloads_bucket_name
+):
+    object_properties = _create_generic_object(
+        boto3.client("sts"), deployment_aws_account_id, downloads_bucket_name
+    )
+    yield object_properties
+    created_object = object_properties["object"]
+    # since this is a versioned bucket, to completely delete the object, the version ID needs to be specified
+    created_object.delete(VersionId=created_object.version_id)
+
+
+def test_When_admin_account_assumes_marketing_role__Then_an_object_can_be_created_and_deleted_in_the_downloads_bucket(
+    tf_workspace_name, deployment_tier, deployment_aws_account_id, downloads_bucket_name
+):
+    print(f"Workspace name in pytest: {tf_workspace_name}")  # allow-print
+    print(f"Determined deployment tier: {deployment_tier}")  # allow-print
+    client = boto3.client("sts")
+    account_id = client.get_caller_identity()["Account"]
+    # confirm that the client being used to create the object is the admin account
+    assert account_id == "424924102580"
+    object_properties = _create_generic_object(
+        client, deployment_aws_account_id, downloads_bucket_name
+    )
+    created_object = object_properties["object"]
+    expected_key = object_properties["key"]
+    assert created_object.key == expected_key
+
+    # since this is a versioned bucket, to completely delete the object, the version ID needs to be specified
+    created_object.delete(VersionId=created_object.version_id)
+
+
+def test_Given_an_object_is_in_the_downloads_bucket__When_a_generic_http_request_is_sent_for_the_object__Then_it_can_be_accessed_since_the_bucket_is_configured_for_public_access(
```

Comment:
this name is not readable

Reply:
are you saying the whole Given/When/Then naming convention you think is too long and unreadable using snake case? (if so I can just revert to more standard test naming convention)  

or that specifically the way I wrote this is confusing? (if so, could you provide an example of a way you think it is clearer / more readable)

Reply:
I think it's too long, if there are more than a few tests with long names like this it would be difficult to maintain. I think if it needs a description about what the test is doing it could be in a doc-string, but keep the name of the test short. 

There are pytest plugins that for doing BDD if you wanted to go that route.

Reply:
let me know what you think about this format. otherwise I can just revert back to regular test naming for the time being